packages/my-queries-storage/src/compass-pipeline-storage.ts

```
@@ -42,24 +41,32 @@ export class CompassPipelineStorage implements PipelineStorage {
       : this.create(attributes));
   }
 
-  private async create(data: Omit<SavedPipeline, 'lastModified'>) {
-    await this.userData.write(data.id, {
-      ...data,
-      lastModified: Date.now(),
-    });
-    return await this.loadOne(data.id);
+  async create(data: Omit<SavedPipeline, 'lastModified'>): Promise<boolean> {
+    try {
+      await this.userData.write(data.id, {
+        ...data,
+        lastModified: Date.now(),
+      });
+      return true;
+    } catch {
+      return false;
```

Here and in other places, would it make sense to instead be throwing these errors? That way the places that use these storage methods can surface errors to users when they happen. Otherwise folks wouldn't know when something failed to write/load/update without checking the logs.

---

I may be wrong, but when I follow this up to the UI calls, the UI ignores return values. if we let the errors bubble up, the UI would crash (to my understanding) with unhandled promise rejections, right?

do you want me to add proper error handling in the UI layer as well? Just not clear if that change is within the scope of this PR or not, happy to do so that if that's what you're suggesting

---

I think this would be a breaking change for a lot of compass, yeah... probably not something we need to immediately address with these changes even though we know that these older compass codepaths were a bit more lenient with error handling